include/deal.II/lac/matrix_scaling.h

```diff
@@ -0,0 +1,283 @@
+// ------------------------------------------------------------------------
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+// Copyright (C) 2001 - 2025 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// Part of the source code is dual licensed under Apache-2.0 WITH
+// LLVM-exception OR LGPL-2.1-or-later. Detailed license information
+// governing the source code and code contributions can be found in
+// LICENSE.md and CONTRIBUTING.md at the top level directory of deal.II.
+//
+// ------------------------------------------------------------------------
+
+#ifndef dealii_matrix_scaling_h
+#define dealii_matrix_scaling_h
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/enable_observer_pointer.h>
+#include <deal.II/base/exceptions.h>
+
+#include <deal.II/lac/full_matrix.h>
+#include <deal.II/lac/sparse_matrix.h>
+#include <deal.II/lac/vector.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * This class provides access to various matrix scaling algorithms. The
+ * scaling algorithms scale the matrix in place by the action of two diagonal
+ * matrices: one acting on the rows and one acting on the columns. The diagonals
+ * of the two scaling matrices applied on the last scaled matrix are stored in
+ * the class as vectors. The algorithms implemented here are the Sinkhorn-Knopp
+ * algorithm described in this <a
+ * href="https://doi.org/10.1137/060659624">article</a> and the scaling
+ * algorithm that preserves symmetry described in this <a
+ * href="https://doi.org/10.1137/110825753">article</a>.
```

Comment:
I think a better approach would be if you added these references to `doc/doxygen/references.bib` and then cited them here via `@cite`.